lib/internal/crypto/x509.js

```diff
@@ -208,7 +258,7 @@ class X509Certificate {
       value = this[kHandle].validFrom();
       this[kInternalState].set('validFrom', value);
     }
-    return value;
+    return parseASN1Time(value);
```

Comment:
I'm definitely -1 on changing the type of both `validFrom` and `validTo` like this. Not only does it make this a semver-major change, it has impact outside of node.js in runtimes that implement node.js compat here. Instead, we should implement separate getters specifically for this pattern, e.g. `cert.validToDate` and `cert.validFromDate`, keeping the existing signatures of `cert.validTo` and `cert.validFrom` alone.